### ISSUE TITLE
Mirror timing port

### ISSUE THREAD

**Issue by FD-** [dsafa22/AirplayServer#6]
Are you sure the mirroring has its own timing port? I couldn't find this in a capture of traffic between an AppleTV and an iPad. In response to the SETUP packet that contains the StreamConnectionID, the server just responds with a data port.

Overall, this is how I understand SETUP:
* The first call sets up ports that are applicable to all further streams: Timing and Event
* Further calls can be distinguished by their type: 110 means video and 96 means audio
   * Audio has data and control (over UDP)
   * Video has just data (over TCP)

Setup 1:

Client->Server:
```
{
	...
	timingPort = 59205;
	ekey = <omitted>;
	eiv = <omitted>;
        ...
}
```

Server->Client:
```
{
	timingPort = 59750;
	eventPort = 49210;
}
```

Setup 2:

Client->Server
```
{
	streams = (
               {
			type = 110;
                        ...
			streamConnectionID = omitted;
		},
	);
}
```

Server->Client

```
{
	streams = (
		{
			type = 110;
			dataPort = 49211;
		},
	);
}
```

Setup 3:

Client->Server

```
{
	streams = (
		{
			...
			controlPort = 50439;
                        ...
			type = 96;
		},
	);
}
```

Server->Client

```
{
	streams = (
		{
			controlPort = 61818;
			type = 96;
			dataPort = 63356;
		},
	);
}
```

**Comment by FD-** [comment]
BTW, I'm willing to contribute a pull request for this, but not before I know what license I can contribute under. See #3.

**Comment by dsafa22** [comment]
Video and audio have their own time ports for audio-picture synchronization.But audio-picture synchronization not yet implemented in this project.Can you provide the capture between an AppleTV and an iPad? Such as Wireshark capture which I can open?

**Comment by FD-** [comment]
The whole capture is almost 500MB big, but I can extract parts of it if you tell me what exactly you're interesting in.

**Comment by dsafa22** [comment]
Only three "SETUP" requests and responses.

**Comment by FD-** [comment]
I don't want the capture to be publicly accessible. Do you have an email address I can send it to?

**Comment by dsafa22** [comment]
Email:229340646@qq.com

**Comment by FD-** [comment]
I sent you an email.

**Comment by FD-** [comment]
It seems like there is one global timing port on each side that is then used by both audio and video.

**Comment by dsafa22** [comment]
Yes.The capture show the right way.I'll try the new way, and if it's right, I'll submit it.

**Comment by dsafa22** [comment]
At present, this system can also work properly. What makes me wonder is whether Airplay supports two sets of logic?

**Comment by FD-** [comment]
The new way is working properly for me. I have already implemented it locally. I would have contributed it to this project if the license (see #3) was clarified.

**Comment by dsafa22** [comment]
@FD- Do you still keep the original capture? I want all UDP packages. Can you send them to my email? Thanks.

**Comment by FD-** [comment]
All UDP packets are still almost 100MB big. Do you probably just need the UDP traffic on a specific port?

**Comment by FD-** [comment]
BTW, I found if we handle the SETUP calls by their content (whether streams are present and what type they have), we can add support for iOS 9, which doesn't work atm. Once I've uploaded my Raspberry Pi port, I can show you how exactly I did it.

**Comment by dsafa22** [comment]
@FD- You can delete most of the audio data packet.

**Comment by dsafa22** [comment]
@FD- I know how airplay 1 works, but I just don't want to support it.

**Comment by FD-** [comment]
The protocol that iOS 9 uses is actually mostly the same as implemented in your project. Also, it seems the protocol you implemented in your project is a legacy protocol, as used by the AppleTV 3. AppleTV 4 and newer (running tvOS 12) works a bit different in that all communication on the main HTTP connection is encrypted after the fp-setup handshake. I investigated the AirPlay version history and will publish my findings in my repo. Is there anything specific that convinced you your protocol is AirPlay 2?

**Comment by FD-** [comment]
I just published my research at https://github.com/FD-/RPiPlay#airplay-protocol-versions

**Comment by dsafa22** [comment]
@FD- I will study your project.
